cmd/create/operatorroles/cmd.go

```diff
@@ -197,40 +208,253 @@ func run(cmd *cobra.Command, argv []string) error {
 			os.Exit(1)
 		}
 	}
+	policies, err := r.OCMClient.GetPolicies("OperatorRole")
+	if err != nil {
+		r.Reporter.Errorf("Expected a valid role creation mode: %s", err)
+		os.Exit(1)
+	}
 
-	roleName, err := aws.GetInstallerAccountRoleName(cluster)
+	defaultPolicyVersion, err := r.OCMClient.GetDefaultVersion()
 	if err != nil {
-		r.Reporter.Errorf("Expected parsing role account role '%s': %v", cluster.AWS().STS().RoleARN(), err)
+		r.Reporter.Errorf("Error getting latest default version: %s", err)
 		os.Exit(1)
 	}
 
-	path, err := getPathFromInstallerRole(cluster)
+	if args.prefix != "" {
+		return handleOperatorRoleCreationByPrefix(r, env, permissionsBoundary,
+			mode, policies, defaultPolicyVersion)
+	}
+	return handleOperatorRoleCreationByClusterKey(r, env, permissionsBoundary,
+		mode, policies, defaultPolicyVersion)
+}
+
+func handleOperatorRoleCreationByPrefix(r *rosa.Runtime, env string,
```

Comment:
This function is very long, should we break it into smaller functions to improve readability?